packages/gatsby-source-contentful/src/normalize.js

```diff
@@ -54,17 +53,25 @@ const fixId = id => {
 }
 exports.fixId = fixId
 
-exports.fixIds = object => {
-  const out = deepMap(object, (v, k) => (k === `id` ? fixId(v) : v))
+const fixIds = object =>
+  _.mapValues(object, (val, key) => {
+    if (key === `sys`) {
+      val = {
+        ...val,
+        id: fixId(val.id),
+        contentful_id: val.id,
+      }
+    }
 
-  return {
-    ...out,
-    sys: {
-      ...out.sys,
-      contentful_id: object.sys.id,
-    },
-  }
-}
+    if (_.isArray(val)) {
+      return _.toArray(fixIds(val))
+    }
+    if (_.isPlainObject(val)) {
+      return fixIds(val)
+    }
```

Comment:
This should account for circular references otherwise we'll get stack overflow errors. Probably also need a test with circular references.